### DISCUSSION TITLE
Patience reset in EarlyStopping once loss has improved

### DISCUSSION THREAD

**Discussion by danielkorth** [Lightning-AI/pytorch-lightning#7849] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hey, 
im wondering whether the patience parameter is or can be reset once we have started improving again. Reading the docs it sounds like the patience parameter is the absolute number of steps the loss is allowed to not decrease. Is it possible to have it such that the patience counter is reset to its original value whenever we have improved?

**Comment by awaelchli** [comment, marked as answer]
This is exactly how patience is supposed to work. As you can see [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/7b531ac7ac86141237f758a7e7446930964b91a9/pytorch_lightning/callbacks/early_stopping.py#L230), the counter resets to 0 upon improvement.
